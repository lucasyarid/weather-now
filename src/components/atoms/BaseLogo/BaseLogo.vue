<template>
  <figure class="BaseLogo">
    <img class="BaseLogo__img" :src="src" alt="WEATHERNOW Logo" />
  </figure>
</template>

<script>
export default {
  name: "BaseLogo",
  props: {
    variation: {
      type: String,
      default: "default"
    }
  },
  computed: {
    src() {
      return require("@/assets/images/logo/" + this.variation + ".svg");
    }
  }
};
</script>

<style lang="scss" scoped>
.BaseLogo {
  margin: 0;
  width: 100%;
  height: fit-content;
  &__img {
    width: 100%;
  }
}
</style>
