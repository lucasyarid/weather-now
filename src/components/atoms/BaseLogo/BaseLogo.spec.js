import { shallowMount } from "@vue/test-utils";
import BaseLogo from "./BaseLogo";

describe("BaseLogo", () => {
  const WRAPPER = shallowMount(BaseLogo);

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
