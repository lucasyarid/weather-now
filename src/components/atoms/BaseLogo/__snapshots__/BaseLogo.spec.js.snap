// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLogo Componente n√£o alterado 1`] = `"<figure class=\\"BaseLogo\\"><img src=\\"default.svg\\" alt=\\"WEATHERNOW Logo\\" class=\\"BaseLogo__img\\"></figure>"`;
